joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/callgraph/StaticCallLinker.scala

```diff
@@ -1,64 +1,39 @@
 package io.joern.x2cpg.passes.callgraph
 
 import io.shiftleft.codepropertygraph.Cpg
-import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.nodes.*
 import io.shiftleft.codepropertygraph.generated.{DispatchTypes, EdgeTypes}
-import io.shiftleft.passes.CpgPass
-import io.shiftleft.semanticcpg.language._
+import io.shiftleft.passes.{ConcurrentWriterCpgPass, ForkJoinParallelCpgPass}
+import io.shiftleft.semanticcpg.language.*
 import org.slf4j.{Logger, LoggerFactory}
 
-import scala.collection.mutable
+class StaticCallLinker(cpg: Cpg) extends ForkJoinParallelCpgPass[Seq[Call]](cpg) {
```

Comment:
Also, why not use `Method` as the part? If method is the part you can query `cpg.call.methodFullNameExact(method.fullName)` exactly once for each method, whereas you may query your existing `cpg.method.fullNameExact(call.methodFullName)` potentially multiple times. This would avoid the need for a batch size at all.

Reply:
While a batch size would be more consistent for each thread, does it outweigh the cost of the additional graph reads? I suppose it's debatable

Reply:
I would say so yes, maybe do the check on `finish`, example of a check you could add to detect multiple methods for one static call.
```
override def finish(): Unit = {
   val invalidCallRelations = cpg.call.count(x => x.dispatchType == STATIC_DISPATCH && x.callee.size > 1)
   if (invalidCallRelations > 0) {
      logger.warn(s"Invalid CPG structure detected, as $invalidCallRelations static calls have multiple callees")
   }
}
```